compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -944,7 +971,9 @@ convertExpr env0 e = do
             let fldIndex = fromJust (elemIndex x vs)
             let fldName = fldNames !! fldIndex
             recTyp <- convertType env (varType bind)
-            pure $ ERecProj (fromTCon recTyp) fldName scrutinee' `ETmApp` EUnit
+            if envLfVersion env `supports` featureTypeSynonyms
+              then pure $ EStructProj fldName scrutinee' `ETmApp` EUnit
+              else pure $ ERecProj (fromTCon recTyp) fldName scrutinee' `ETmApp` EUnit
```

Comment:
I think you can replace this `if` expression by using the helper function,

```Haskell
pure $ mkDictProj env (fromTCon recTyp) fldName scrutinee' `ETmApp` EUnit
```

Reply:
good catch, fixed. Thank you!